config/reader.go

```diff
@@ -596,6 +627,13 @@ func processConfig(unprocessedConfig *Config, fromCloud bool, logger logging.Log
 		}
 	}
 
+	// add additional environment vars to modules
+	// adding them here ensures that if the parsed API key changes, the module will be restarted with the updated environment.
```

Comment:
> the module will be restarted

Where is that happening?

Reply:
this happens when we diff the config (the env being different on the module config will cause a reconfigure on the module), so more of just utilizing the existing mechanisms to make sure the module is running in the most up to date environment

Reply:
Cool that makes sense; thanks.